### ISSUE TITLE
Subgroups are flattened when Sources paths contain ".." in YAML file

### ISSUE THREAD

**Issue by nzhuk** [yonaskolb/XcodeGen#525]
xcodegen 2.2.0

Test setup:

```
.
├── Sources
│   ├── Subfolder
│   │   └── file2.swift
│   └── file1.swift
└── spec
    └── project.yml
```

`spec/project.yml`:
```
name: MyProject
targets:
  MyFramework:
    type: framework
    platform: iOS
    sources:
      - path: ../Sources
```

Command: `xcodegen --spec spec/project.yml`

`MyProject.xcodeproj` looks like this:

![screenshot 2019-02-22 10 43 46](https://user-images.githubusercontent.com/294154/53230169-c16e5480-368e-11e9-9c58-25de94d9986c.png)

Current: `Subfolder` is not a subgroup of `Sources` group
Expected: `Subfolder` is a subgroup of `Sources` group

Note: This seems to be triggered when path of Sources in YAML file contains at least one `../`

**Comment by yonaskolb** [comment]
Thanks for the report @nzhuk. This will most likely be fixed by #524

**Comment by sroebert** [comment]
I just checked, but it does not seem to solve this issue.

**Comment by alejandro-isaza** [comment]
As an alternative I tried moving the project.yml file one level up and using `codegen --project subfolder` but then the references in the project are broken: `
Build input file cannot be found: '...'`

**Comment by yonaskolb** [comment]
Can you try again in Xcode 2.5.0, now that it includes #550

**Comment by nzhuk** [comment]
@yonaskolb The reported issue seems to be fixed in 2.5.0.

`xcodegen --spec spec/project.yml` produces correct project:

![Screenshot 2019-04-10 12 42 32](https://user-images.githubusercontent.com/294154/55868930-2114b680-5b8e-11e9-8ac7-8be9a4553180.png)


However, if the Xcode project is generated to the parent directory of `spec` directory, the project is still broken (probably another issue, not sure where it's tracked):

`xcodegen --spec spec/project.yml -p .`:

![Screenshot 2019-04-10 12 44 13](https://user-images.githubusercontent.com/294154/55869040-5faa7100-5b8e-11e9-94e5-64e02cfa5df2.png)

(here, "Sources" group refers to invalid folder, thus "test.swift" is not found)

**Comment by yonaskolb** [comment]
@nzhuk, great I'll close this issue then. I believe the other problem you mention has come up before. I think there are issues with the `--project` argument. Would you mind opening a new issue, and we can see if we can get it fixed.
